javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodExampleMapper.java

```diff
@@ -32,7 +37,18 @@ public ProxyMethodExample map(Object xmsExample) {
             }
             String xmsOriginalFile = (String) ((Map<String, Object>) xmsExample).get("x-ms-original-file");
             builder.originalFile(xmsOriginalFile);
+            if (exampleWrapper.getOperationId() != null) {
+                builder.codeSnippetIdentifier(buildCodeSnippetIdentifier(exampleWrapper.getOperationId(), exampleWrapper.getExampleName()));
+            }
         }
         return builder.build();
     }
+
+    private String buildCodeSnippetIdentifier(String operationId, String exampleName) {
+        return String.format("%s.generated.%s.%s", JavaSettings.getInstance().getPackage(), getValidName(operationId), getValidName(exampleName)).toLowerCase(Locale.ROOT);
+    }
+
+    private String getValidName(String exampleName) {
+        return CodeNamer.getValidName(exampleName).replace("_", "");
```

Comment:
Probably replace `-` as well? Or just use `CodeNamer.getValidName(name)` without `allowedCharacters`?

Reply:
Yes, I used `CodeNamer.getValidName(name)` without `allowedCharacters`, and invalid characters will be replaced by `_` :
https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/util/CodeNamer.java#L356-L362
so I do an extra work of removing `_`.

Reply:
BODY parameter will not show up here because it is filtered out in the previous steps:
https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java#L39-L51